src/coreclr/src/gc/gcinterface.dac.h

```diff
@@ -17,6 +17,8 @@
 #define MAX_EXPAND_MECHANISMS_COUNT     6
 #define MAX_GC_MECHANISM_BITS_COUNT     2
 #define MAX_GLOBAL_GC_MECHANISMS_COUNT  6
+
+// for backward compatibility the number of generations for DAC must stay 4
```

Comment:
I don't think this is the case - you can make it work, it's just work that we need to do. it would be very odd if we just forever only process 4 generations. please talk to the diagnostics folks about this.

Reply:
It looks like the SOS DAC api (in dacprivate.h starting at line 712) assume that there are only 4 generations. The DacpGcHeapDetails request and DacpGenerationAllocData assume 4 generations. SOS and probably CLRMD use this API so we need to come up with a backward compatible way to support this.  Like a new Request (DacpGcHeapDetails2 ?) that the 5.0 DAC supports.

Reply:
Not that this is impossible, but it is not something to be changed lightly. It is a breaking change. 
It is also not clear whether if it is better to introduce a new generation for POH or after, or just fold everything UOH together.
There is a work item for this - [https://github.com/dotnet/coreclr/issues/27693](https://github.com/dotnet/coreclr/issues/27693) 


---
In reply to: [368190184](https://github.com/dotnet/runtime/pull/1688#discussion_r368190184) [](ancestors = 368190184)

Reply:
I certainly wasn't suggesting changing this lightly. I was suggesting we should not have an incorrect comment. we could say that it will require work to change this. feel free to open an issue for changing it in the future.

Reply:
I will remove the comment. It does not make a lot of sense in this change, and related work is tracked in a workitem.